crates/apub/apub/src/collections/community_outbox.rs

```diff
@@ -37,13 +35,15 @@ impl Collection for ApubCommunityOutbox {
   async fn read_local(owner: &Self::Owner, data: &Data<Self::DataType>) -> LemmyResult<Self::Kind> {
     let site = Site::read_local(&mut data.pool()).await?;
 
-    let post_views = PostQuery {
-      community_id: Some(owner.id),
-      sort: Some(PostSortType::New),
-      limit: Some(FETCH_LIMIT_MAX.try_into()?),
-      ..Default::default()
-    }
-    .list(&site, &mut data.pool())
+    let post_views = Box::pin(
+      PostQuery {
+        community_id: Some(owner.id),
+        sort: Some(PostSortType::New),
+        limit: Some(FETCH_LIMIT_MAX.try_into()?),
+        ..Default::default()
+      }
+      .list(&site, &mut data.pool()),
+    )
     .await?;
```

Comment:
This small change reduced compilation time by 10s. There is a [related lint](https://rust-lang.github.io/rust-clippy/master/index.html#large_futures) but it doesnt mark this as a large future,  not even when `future-size-threshold` is set to a much lower value than the default. Also adding Box::pin to other places where PostQuery is used doesnt make any difference. All very strange.